### PR TITLE
Roadmap update

### PR DESCRIPTION
**Pull request by brunoolivieri**
I took the liberty to propose that relying on the un-conference discussion. This is a copy from the spreadsheet used during the un-conference. 

It is roughly alphabetically sorted order but pushing vehicles up and merge some sections based only in names to domain area (ie miscellaneous)

I need some help with 2019 items once I am not sure if they were finished or not.

### CODE REVIEW

File: dev/source/docs/roadmap.rst [hunk 1]
```diff
@@ -1,120 +1,118 @@
...
-  - Handle EKF reset for position correctly
+Rover & Boat
+------------
 
-- AutoTune
+- S-Curve navigation (`video from 2019 un-conference <https://www.youtube.com/watch?v=LHq5o9zgNWk>`__) (Leonard/Randy)
+- Stick Mixing in Auto (TomP)
+- Underwater mapping with scanning sonar (Randy/PeterB)
+- Precision Docking (using marker onshore) (Randy)
+- high speed sailing inprovements (Randy)
```
Comment (rmackay9) [inline dev/source/docs/roadmap.rst:64]: These three items (high speeds sailing, roll and pitch control and AIS) are from Peter Hall.

File: dev/source/docs/roadmap.rst [hunk 2]
```diff
@@ -1,120 +1,118 @@
...
-- APSync to support connecting to external Wifi access point (Randy/Peter)
+EKF (PaulR / Randy)
+-------------------
 
-Documentation (Randy / PierreK)
--------------------------------
+- Bring the missing features from EKF2 into EKF3 to remove EKF2 (Paul)
+- EKF3 External Nav support
+- Fixing replay (PeterB)
+- Kill EKF2 after feature capability of EKF3
```
Comment (rmackay9) [inline dev/source/docs/roadmap.rst:89]: let's change this line with "Kill" to be, "Set EKF3 as the default"

File: dev/source/docs/roadmap.rst [hunk 3]
```diff
@@ -1,120 +1,118 @@
...
+- Possibly switch to copter PID controllers
```
Comment (rmackay9) [inline dev/source/docs/roadmap.rst:90]: "Possibly switch to copter PID controllers" should be moved to the plane section

File: dev/source/docs/roadmap.rst [hunk 4]
```diff
@@ -1,120 +1,118 @@
...
+- Stick Mixing in Auto (TomP)
+- Underwater mapping with scanning sonar (Randy/PeterB)
+- Precision Docking (using marker onshore) (Randy)
+- high speed sailing inprovements (Randy)
+- Roll and pitch control for boats (Randy)
+- AIS (i.e., ADSB for boats) (Randy)
 
-  - Add tuning type to PID object
 
 Trad Heli (Bill Geyer, Chris Olson)
```
Comment (rmackay9) [inline dev/source/docs/roadmap.rst:69]: Let's change the title to be just, "Trad Heli (Bill Geyer)".

File: dev/source/docs/roadmap.rst [hunk 5]
```diff
@@ -125,16 +123,95 @@ Hardware (PhilipR, JeffW)
...
-- Support adding information about messages and fields to Onboard logs
+Non-GPS Navigation, Avoidance and Path Planning (Randy / Patrick Poirier)
+-------------------------------------------------------------------------
+
+- 3D Bendy Ruler & Object Database (Randy)
+- Add support for Intel RealSense D435 (ROS & non-ROS) (Patrick, GSoC)
+- Bendy Ruler try re-implementing bendy ruler as Lua (Tridge)
+- Copter & Rover back away from objects (Randy)
+- Improve Off-Board SLAM and Object Avoidance (:ref:`dev wiki link<ros-cartographer-slam>`) (Randy, Jaime)
+- Improve reliability of T265 integration (reset handling, failsafe testing) (Randy, Jaime)
```
Comment (rmackay9) [inline dev/source/docs/roadmap.rst:135]: Let's add Patrick Poirier to the list of names working on this.

File: dev/source/docs/roadmap.rst [hunk 6]
```diff
@@ -125,16 +123,95 @@ Hardware (PhilipR, JeffW)
...
+- T265 image output to external VO (Tridge)
+
+
+Scripting (MichaelDB)
+---------------------
+
+- All mission features available in Lua (MichaelDB)
+- Figure flight mode in Plane as a script (MichaelDB)
+- Scripts creation parameter trees (MichaelDB)
+- Sprayer moved out of C++, into Lua (MichaelDB)
```
Comment (rmackay9) [inline dev/source/docs/roadmap.rst:146]: I think this item might have been mine so let's change the owner to "Randy"

File: dev/source/docs/roadmap.rst [hunk 7]
```diff
@@ -125,16 +123,95 @@ Hardware (PhilipR, JeffW)
...
+- Improve Off-Board SLAM and Object Avoidance (:ref:`dev wiki link<ros-cartographer-slam>`) (Randy, Jaime)
+- Improve reliability of T265 integration (reset handling, failsafe testing) (Randy, Jaime)
+- Seamless GPS <-> Non-GPS transitioning (Randy)
+- T265 image output to external VO (Tridge)
+
+
+Scripting (MichaelDB)
+---------------------
+
+- All mission features available in Lua (MichaelDB)
```
Comment (rmackay9) [inline dev/source/docs/roadmap.rst:143]: Let's make the owner of this be "MichaelDB, Randy"

**Comment by rmackay9** [comment]
Thanks very much for this. I've added a few corrections but you've saved me a ton of time by putting this together!

File: dev/source/docs/roadmap.rst (hunk 1 above)
Comment (brunoolivieri) [reply dev/source/docs/roadmap.rst]: ok, done.

File: dev/source/docs/roadmap.rst (hunk 2 above)
Comment (brunoolivieri) [reply dev/source/docs/roadmap.rst]: ok, done.

File: dev/source/docs/roadmap.rst (hunk 3 above)
Comment (brunoolivieri) [reply dev/source/docs/roadmap.rst]: ok, done.

File: dev/source/docs/roadmap.rst (hunk 4 above)
Comment (brunoolivieri) [reply dev/source/docs/roadmap.rst]: ok, done.

File: dev/source/docs/roadmap.rst (hunk 5 above)
Comment (brunoolivieri) [reply dev/source/docs/roadmap.rst]: ok, done.

File: dev/source/docs/roadmap.rst (hunk 7 above)
Comment (brunoolivieri) [reply dev/source/docs/roadmap.rst:143]: ok, done.

File: dev/source/docs/roadmap.rst (hunk 6 above)
Comment (brunoolivieri) [reply dev/source/docs/roadmap.rst]: ok, done.

**Comment by rmackay9** [comment]
@brunoolivieri, thanks very much!  really appreciate it!

**Comment by brunoolivieri** [comment]
@rmackay9 I am glad to help but I guess it is not ready to merge yet. I need some help with 2019 section. Could help me selecting what is complete or not?

**Pull request merged**